.github/PULL_REQUEST_TEMPLATE

```diff
@@ -1,10 +1,32 @@
-## What changes were proposed in this pull request?
+<!--  Thanks for sending a pull request!  Here are some tips for you:
+1. If this is your first time, please read our contributor guidelines: https://spark.apache.org/contributing.html
+2. Ensure you have added or run the appropriate tests for your PR: https://spark.apache.org/developer-tools.html
+3. Understand the relationship beteween JIRAs to file issues and PRs to propose some changes for them, as documented in guidelines.
+4. If the PR is unfinished, add '[WIP]' in your PR title, e.g., '[WIP][SPARK-XXXX] Your PR title ...'
+5. Update your PR description whenever it is needed.
+6. Provide a minimised example to reproduce the issue if possible.
+-->
 
-(Please fill in changes proposed in this fix)
+### What changes were proposed in this pull request?
 
-## How was this patch tested?
+### Why changes are needed?
```

Comment:
What we expect in these two sections: `### What changes were proposed in this pull request?` and `### Why changes are needed?`  ? Should we combine them?

Reply:
We can combine but I expected to explicitly mention those two separately:

```
### What changes were proposed in this pull request?

Describe how this PR fixes the JIRA. For instance, A calls B but B has to be C blah blah.

### Why changes are needed?

e.g.,
1. This API has been requested in user mailing list multiple times. See http://... , http:// ...
2. It follows those references to make Spark's behaviour reasonable.
3. This is a bug fix that introduced by SPARK-XXXXX. It should work as expected.
```

Reply:
Because nowadays Spark's being conservative when it adds some APIs and we need to know why such APIs are needed, for instance.

Reply:
and .. they are separate two sections to avoid people ignore one of both questions. @gengliangwang pointed out similar one and I answered at https://github.com/apache/spark/pull/25310#discussion_r309149015.

If you guys prefer combining, I don't mind but I wanted to say why I separated them.

Reply:
Do we add content in your comment to the template? I guess we need to have some explanation on both sections.

Reply:
will clarify it.